### ISSUE TITLE
Setting `resourceAreaHeaderContent` overrides the value of `headerContent` is resource columns

### ISSUE THREAD

**Issue by tn-bouvet** [fullcalendar/fullcalendar#7153]
### Reduced Test Case

https://stackblitz.com/edit/github-6b7aj5?file=src/index.jsx

### Do you understand that if a reduced test case is not provided, we will intentionally delay triaging of your ticket?

- [X] I understand

### Which connector are you using (React/Angular/etc)?

React

### Bug Description

See stackblitz project. When `resourceAreaHeaderContent` is provided, the calendar overwrites the `headerContent` in the `resourceAreaColumns` prop. 

Also, while the [docs](https://fullcalendar.io/docs/resource-area-header-render-hooks) state that the default resource area header should be "Resources", no such header appears as long as `resourceAreaColumns` are defined. 

I have checked without the React connector, and this seems to work in that case. See [codepen](https://codepen.io/tn-bouvet/pen/LYBzxrQ?editors=0010). 

Desired outcome (taken without react connector):
![image](https://user-images.githubusercontent.com/88835417/212901231-a245bc10-4964-44fc-bcec-668b2a8547c1.png)

Result when `resourceAreaHeaderContent` is provided:
![image](https://user-images.githubusercontent.com/88835417/212901419-d3c9cb5f-1ac2-4975-97bd-8bdd687d2e67.png)

Result when `resourceAreaHeaderContent` is not provided:
![image](https://user-images.githubusercontent.com/88835417/212901554-fbb9be06-cd17-4f1c-8542-fbc17521426e.png)

**Comment by lukassiarny** [comment]
The cellContent with resourceAreaColumns also dosen't work correctly. It works when is set to string, but when you return a JSX it returns nothing.

I just added cellContent to the test case above:
This doesn't work:
```
          resourceAreaColumns={[
            { field: 'num', headerContent: () => 'Number' },
            { field: 'name', headerContent: 'Name', cellContent: () => <span>'cellContent'</span> },
          ]}
```
This works:
```
          resourceAreaColumns={[
            { field: 'num', headerContent: () => 'Number' },
            { field: 'name', headerContent: 'Name', cellContent: () => 'cellContent' },
          ]}
```

**Comment by acerix** [comment]
Thanks, I can confirme this is a bug specfic to React.

Since `resourceAreaHeaderContent` is getting called for each header, as a workaround, you can add a counter and return the correct header content like this:

https://stackblitz.com/edit/github-6b7aj5-sgfkgx?file=src%2FDemoApp.jsx

**Comment by arshaw** [comment]
Thanks for this bug report. I'll fix it and release it in a minor release before the end of next week (Jan 27). Sorry for the troubles.

**Comment by arshaw** [comment]
This is fixed in v6.1.0!

- blog post: https://fullcalendar.io/blog/2023/01/multimonth-and-more
- github release: https://github.com/fullcalendar/fullcalendar/releases/tag/v6.1.0
- updated repro: https://stackblitz.com/edit/github-6b7aj5-nebprp?file=package.json
